project_national/sdr_upgrades_furnace.yml

```diff
@@ -0,0 +1,325 @@
+    reporting_measures:
+      - measure_dir_name: QOIReport
+      - measure_dir_name: ServerDirectoryCleanup # tack this on so we can save some disk space (when using debug: true)
+
+    # Files to keep
+    server_directory_cleanup:
+      retain_in_idf: true
+      retain_schedules_csv: true
+      retain_in_osm: true
+
+    debug: true
+
+references:
+  remove_nonducted_high_rise_shared: &remove_nonducted_high_rise_shared
+    - HVAC Has Ducts|Yes
+    - not: Heating Fuel|Wood
+    - not: Heating Fuel|Other Fuel
+    - not: Heating Fuel|None
+    - not:
+      - or:
+        - HVAC Shared Efficiencies|Boiler Baseboards Heating Only, Electricity
+        - HVAC Shared Efficiencies|Boiler Baseboards Heating Only, Fuel
+        - HVAC Shared Efficiencies|Fan Coil Heating and Cooling, Electricity
+        - HVAC Shared Efficiencies|Fan Coil Heating and Cooling, Fuel
+        - HVAC Shared Efficiencies|Fan Coil Cooling Only
+      - and:
+        - Geometry Building Type Height|Multi-Family with 5+ Units, 8+ Stories
+
+
+upgrades:
+  - upgrade_name: ng_95_afue_for_all
+    package_apply_logic: *remove_nonducted_high_rise_shared
+    options:
+        ### -- Natural Gas Furnace AFUE 95% applies to all dwellings with ducts
+      - option: HVAC Heating Efficiency|Fuel Furnace, 95% AFUE
+        # no apply logic on purpose since package_apply_logic handles it
+        lifetime: 15
+      - option: HVAC Heating Type And Fuel|Natural Gas Fuel Furnace
+        apply_logic: 
+          - not: Heating Fuel|Natural Gas
+
+  - upgrade_name: prop_fo_furnace
+    package_apply_logic: *remove_nonducted_high_rise_shared
+    options:
+      - option: HVAC Heating Efficiency|Fuel Furnace, 95% AFUE
+        apply_logic: 
+          - Heating Fuel|Propane
+          - or:
+            - HVAC Heating Efficiency|Fuel Furnace, 60% AFUE
+            - HVAC Heating Efficiency|Fuel Furnace, 68% AFUE
+            - HVAC Heating Efficiency|Fuel Furnace, 72% AFUE
+            - HVAC Heating Efficiency|Fuel Furnace, 76% AFUE
+            - HVAC Heating Efficiency|Fuel Furnace, 80% AFUE
+            - HVAC Heating Efficiency|Fuel Furnace, 85% AFUE
+            - HVAC Heating Efficiency|Fuel Furnace, 90% AFUE
+            - HVAC Heating Efficiency|Fuel Furnace, 92.5% AFUE
+            - HVAC Heating Efficiency|Fuel Furnace, 95% AFUE
+        lifetime: 15
+      - option: HVAC Heating Efficiency|Fuel Furnace, 88% AFUE
+        apply_logic: 
+          - Heating Fuel|Fuel Oil
+          - or:
+            - HVAC Heating Efficiency|Fuel Furnace, 60% AFUE
+            - HVAC Heating Efficiency|Fuel Furnace, 68% AFUE
+            - HVAC Heating Efficiency|Fuel Furnace, 72% AFUE
+            - HVAC Heating Efficiency|Fuel Furnace, 76% AFUE
+            - HVAC Heating Efficiency|Fuel Furnace, 80% AFUE
+            - HVAC Heating Efficiency|Fuel Furnace, 85% AFUE
+        lifetime: 15
+
+  - upgrade_name: furnace_ac_ref
+    package_apply_logic: *remove_nonducted_high_rise_shared
```

Comment:
For the package_apply_logic: you want to mirror CCHP's apply logic. 
I would copy over both of these references [here](https://github.com/NREL/resstock/blob/0abb6c5ab265744eb6b35c5d9978ee5b49c97af3/project_national/sdr_upgrades_tmy3.yml#L404-L447) and apply both of them to your `package_appy_logic` here.